crates/bevy_app/src/plugin_group.rs

```diff
@@ -2,6 +2,123 @@ use crate::{App, AppError, Plugin};
 use bevy_utils::{tracing::debug, tracing::warn, TypeIdMap};
 use std::any::TypeId;
 
+/// A macro for generating a well-documented [`PluginGroup`] from a list of [`Plugin`] paths.
+///
+/// Every plugin must implement the [`Default`] trait.
+///
+/// # Example
+///
+/// ```
+/// # use bevy_app::*;
+/// # mod velocity {
+/// #   use bevy_app::*;
+/// #   #[derive(Default)] pub struct VelocityPlugin;
+/// #   impl Plugin for VelocityPlugin { fn build(&self, _: &mut App) {} }
+/// # }
+/// # mod collision { pub mod capsule {
+/// #   use bevy_app::*;
+/// #   #[derive(Default)] pub struct CapsuleCollisionPlugin;
+/// #   impl Plugin for CapsuleCollisionPlugin { fn build(&self, _: &mut App) {} }
+/// # } }
+/// # #[derive(Default)] pub struct TickratePlugin;
+/// # impl Plugin for TickratePlugin { fn build(&self, _: &mut App) {} }
+/// # mod features {
+/// #   use bevy_app::*;
+/// #   #[derive(Default)] pub struct ForcePlugin;
+/// #   impl Plugin for ForcePlugin { fn build(&self, _: &mut App) {} }
+/// # }
+/// # mod web {
+/// #   use bevy_app::*;
+/// #   #[derive(Default)] pub struct WebCompatibilityPlugin;
+/// #   impl Plugin for WebCompatibilityPlugin { fn build(&self, _: &mut App) {} }
+/// # }
+/// # mod internal {
+/// #   use bevy_app::*;
+/// #   #[derive(Default)] pub struct InternalPlugin;
+/// #   impl Plugin for InternalPlugin { fn build(&self, _: &mut App) {} }
+/// # }
+/// plugin_group!(
+///     PhysicsPlugins {
+///         // Due to local ambiguity issues, you have to
+///         // use a colon before the plugin's name.
+///         :TickratePlugin,
+///         // Due to local ambiguity issues, you have to
+///         // use 3 colons for the last part of the path.
+///         collision::capsule:::CapsuleCollisionPlugin,
+///         velocity:::VelocityPlugin,
+///         // Add a documented feature restriction.
+///         #[cfg(feature = "external_forces")]
+///         features:::ForcePlugin,
+///         // You can add any attribute you want like this, but it won't be documented.
+///         #[custom(cfg(target_arch = "wasm32"))]
+///         web:::WebCompatibilityPlugin,
+///         // You can hide plugins from documentation.
+///         // NOTE: Due to macro limitations, you can only use
+///         //       #[doc(hidden)] for plugins at the bottom of the macro.
+///         #[doc(hidden)]
+///         internal:::InternalPlugin
+///     }
+/// );
+/// ```
+#[macro_export]
+macro_rules! plugin_group {
+    (
+        $(#[$group_meta:meta])*
+        $group:ident {
+            $(
+                $(#[cfg(feature = $plugin_feature:literal)])?
+                $(#[custom($plugin_meta:meta)])*
+                $($plugin_path:ident::)* : $plugin_name:ident
+            ),*
+            $(
+                ,$(
+                    #[doc(hidden)]
```

Comment:
This will require all hidden plugins to be at the end of the list. Should this be documented?

Also, creating a plugin group with just hidden plugins will require a comma `,` in order to be valid.

```rust
plugin_group!(
    MyPlugins {
        // Required ","
        ,
        #[doc(hidden)]
        :NonDefaultPlugin
    }
);
```

Reply:
> This will require all hidden plugins to be at the end of the list. Should this be documented?

It is documented.

I also fixed the required comma. (It is no longer required)